### DISCUSSION TITLE
Error when reloading or redeploying my project (random-move-engine for a chess-app)

### DISCUSSION THREAD

**Discussion by marte1337** [neuefische/questions#583] [Web] [answered]
Hi, I´m working at this [user-story](https://github.com/marte1337/capstone-project/issues/4) for my[ capstone-project](https://github.com/marte1337/capstone-project).

Libraries are in use for the [chessboard](https://github.com/Clariity/react-chessboard) and for the [game logic](https://github.com/jhlywa/chess.js) I´m running chess.js in version 0.12.0, since this is the latest version I could get the random-move-engine to work, newer versions are still in beta and seem to work a bit different. (See also the code snippet for a random-move-engine in the react-chessboard documentation.)

When running the project locally, everything works fine - I can play a full game of chess against the random-move-engine. But when I reload the page at any point I get the following error message: 

![Screenshot 2023-04-27 at 10 04 32](https://user-images.githubusercontent.com/54943747/234812723-6fbf6cd9-93f1-420e-bd25-29c7110bd4ce.png)

The vercel-redeployment is also doesn´t work at the moment (it did before), throwing the same error:

![Screenshot 2023-04-27 at 10 45 37](https://user-images.githubusercontent.com/54943747/234813039-10b12c74-86d1-439c-919d-476fec522944.png)

EDIT:

Trying to relocate the "new Chess" function in a variable also doesn´t fix the issue.

![Screenshot 2023-04-27 at 11 20 49](https://user-images.githubusercontent.com/54943747/234819588-769cd8e8-34a8-4e7d-b666-be6fc0d4b5ef.png)

**Comment by NikRoe** [comment]
Hi Marte,

can you send us a Pull Request of the current branch?

Thanks in advance :)

**Reply by marte1337** [reply]
Hi, thanks for the fast reply. Here is the PR: https://github.com/marte1337/capstone-project/pull/7

**Reply by NikRoe** [reply, marked as answer]
Thanks for the PR!

We found a solution for the issue:

You need to set the game to the `new Chess` game within a `useEffect`, so that the state gets set after your components have initially been rendered.
Your code should look like this:

```js
const [game, setGame] = useState(null);

  useEffect(() => {
    setGame(new Chess());
  }, []);
```

Furthermore, you should make the return conditional and dependent on the `game`, since you are working with `game.fen()` here:

```js
return game && <Chessboard position={game.fen()} onPieceDrop={onDrop} />;
```

I hope this helps =)

**Reply by marte1337** [reply]
Works! THANK YOU!!
